src/connector/src/sink/mod.rs

```diff
@@ -372,7 +372,7 @@ pub trait Sink: TryFrom<SinkParam, Error = SinkError> {
     type Coordinator: SinkCommitCoordinator;
 
     /// `user_specified` is the value of `sink_decouple` config.
-    fn is_sink_decouple(_desc: &SinkDesc, user_specified: &SinkDecouple) -> Result<bool> {
+    fn is_sink_decouple(_desc: &mut SinkDesc, user_specified: &SinkDecouple) -> Result<bool> {
```

Comment:
This is so hacky. `is_sink_decouple` should be side-effect free based on the method name. How about filling in the default value commit_checkpoint_interval elsewhere when meeting SinkDecouple:default.